src/frontend/js/components/AddressesManagement/useAddressesManagement.tsx

```diff
@@ -0,0 +1,135 @@
+  const [error, setError] = useState<Maybe<string>>();
+  const [load, setLoad] = useState(false);
+
+  /**
+   * Update the provided address to promote it as main
+   *
+   * @param {Joanie.Address} address
+   */
+  const promoteAddress = (address: Joanie.Address) => {
+    if (!address.is_main) {
+      setLoad(true);
+      setError(undefined);
+      addresses.methods.update(
+        {
+          ...address,
+          is_main: true,
+        },
+        {
+          onSuccess: () => {
+            setLoad(false);
+          },
+          onError: () => {
+            setError(intl.formatMessage(messages.actionPromotion));
+            setLoad(false);
+          },
+        },
+      );
+    }
+  };
+
+  /**
+   * Ask the user to confirm his intention
+   * then make the request to delete the provided address
+   *
+   * @param {Joanie.Address} address
+   * @param {AddressesMutateOptions} options
+   */
+  const removeAddress = (
+    address: Joanie.Address,
+    options?: MutateOptions<void, HttpError, string>,
+  ) => {
+    // eslint-disable-next-line no-alert, no-restricted-globals
+    const sure = confirm(
+      intl.formatMessage(messages.deletionConfirmation, { title: address.title }),
+    );
+    setLoad(true);
+    setError(undefined);
+    if (!address.is_main && sure) {
+      setLoad(true);
+      addresses.methods.delete(address.id, {
+        onError: () => {
+          setError(intl.formatMessage(messages.actionDeletion));
+          setLoad(false);
+        },
+        onSuccess: () => {
+          setLoad(false);
+        },
+        ...options,
+      });
+    }
+  };
+
+  /**
+   * Create a new address according to form values
+   *
+   * @param {Address} data address fields to update
+   * @param {AddressesMutateOptions} options
+   */
+  const createAddress = async (data: Omit<Address, 'id'>, options?: AddressesMutateOptions) => {
+    setError(undefined);
+    setLoad(true);
+    await addresses.methods.create(data, {
+      onError: () => {
+        setError(intl.formatMessage(messages.actionCreation));
+        setLoad(false);
+      },
+      onSuccess: () => {
+        setLoad(false);
+      },
+      ...options,
+    });
+  };
+
+  /**
+   * Update the `editedAddress` with new values provided as argument
+   *
+   * @param {Address} data address fields to update
+   * @param {AddressesMutateOptions} options
+   */
+  const updateAddress = async (data: Address, options?: AddressesMutateOptions) => {
+    setError(undefined);
+    setLoad(true);
+    addresses.methods.update(data, {
+      onSuccess: () => {
+        setLoad(false);
+      },
+      onError: () => {
+        setError(intl.formatMessage(messages.actionUpdate));
+        setLoad(false);
+      },
+      ...options,
+    });
+  };
+
+  return {
+    ...addresses,
+    management: {
+      load,
+      setLoad,
+      error,
+      setError,
+      promoteAddress,
+      removeAddress,
+      createAddress,
+      updateAddress,
+    },
```

Comment:
What do you think to split this in several properties ? 

```suggestion
error,
state: { isLoading },
methods: { setError, setIsLoading, promote, remove, create, update }
```

Reply:
This object extends `useAddresses` I didn't want to mix the attributes & methods from low & high level, WDYT ?